src/test/java/bio/terra/service/filedata/google/firestore/FileOperationTest.java

```diff
@@ -194,6 +244,238 @@ public void fileOperationsTest() throws Exception {
+        BulkLoadArrayRequestModel arrayLoad = new BulkLoadArrayRequestModel()
+            .profileId(profileModel.getId())
+            .loadTag(loadTag)
+            .maxFailedFileLoads(0);
+        for (int i = 0; i < 8; i++) {
+            arrayLoad.addLoadArrayItem(getFileModel(true, i, testId));
+        }
+
+        MvcResult result = connectedOperations.ingestArrayRaw(datasetSummary.getId(), arrayLoad);
+        assertThat("Got bad request", result.getResponse().getStatus(), equalTo(HttpStatus.BAD_REQUEST.value()));
+    }
+
+    private static void checkFileResultFailed(BulkLoadFileResultModel fileResult) {
+        assertNotNull("Error is not null", fileResult.getError());
+        assertNull("FileId is null", fileResult.getFileId());
+        assertThat("State is FAILED", fileResult.getState(), equalTo(BulkLoadFileState.FAILED));
+    }
+
+    private static void checkFileResultSuccess(BulkLoadFileResultModel fileResult) {
+        assertNull("Error is null", fileResult.getError());
+        assertNotNull("FileId is not null", fileResult.getFileId());
+        assertThat("State is SUCCEEDED", fileResult.getState(), equalTo(BulkLoadFileState.SUCCEEDED));
+    }
+
+    private static void checkLoadSummary(BulkLoadResultModel summary,
+                                         String loadTag,
+                                         int total,
+                                         int succeeded,
+                                         int failed,
+                                         int notTried) {
+        assertThat("correct load tag", summary.getLoadTag(), equalTo(loadTag));
+        assertThat("correct total files", summary.getTotalFiles(), equalTo(total));
+        assertThat("correct succeeded files", summary.getSucceededFiles(), equalTo(succeeded));
+        assertThat("correct failed files", summary.getFailedFiles(), equalTo(failed));
+        assertThat("correct notTried files", summary.getNotTriedFiles(), equalTo(notTried));
+    }
+
+    private BulkLoadArrayRequestModel makeSuccessArrayLoad(String tagBase, int startIndex, int fileCount) {
+        String testId = Names.randomizeName("test");
+        String loadTag = tagBase + testId;
+        BulkLoadArrayRequestModel arrayLoad = new BulkLoadArrayRequestModel()
+            .profileId(profileModel.getId())
+            .loadTag(loadTag)
+            .maxFailedFileLoads(0);
+        for (int index = startIndex; index < startIndex + fileCount; index++) {
+            arrayLoad.addLoadArrayItem(getFileModel(true, index, testId));
+        }
+        return arrayLoad;
+    }
+
+    // We have a static array of good paths and bad paths with their associated
+    // target. That lets us build arrays with various numbers of failures and
+    // adjust arrays to "fix" broken loads.
+    private static String[] goodFileSource = new String[]{
+        "gs://jade-testdata/encodetest/files/2016/07/07/1fd31802-0ea3-4b75-961e-2fd9ac27a15c/ENCFF580QIE.bam",
```

Comment:
Out of the scope of this PR, but hard-coding these paths will make it hard for non-Jades to use this code. We should either make this data accessible publicly or have a way for people to specify their own paths.

Reply:
Yes, it is a general problem with our tests. Having a (perhaps better organized) bucket of test data that is public would be a good idea. I wonder if we should use either totally synthetic data or 1000 genome data.